src/config.js

```diff
@@ -82,7 +106,8 @@ export function buildConfig (options: ConfigOptions = {}): Config {
     platform: os.platform(),
     arch: os.arch(),
     bin: cli.bin || 'cli-engine',
-    defaultCommand: cli.defaultCommand || 'help'
+    defaultCommand: cli.defaultCommand || 'help',
+    skipAnalytics: false
```

Comment:
I was unable to do this because skipAnalytics requires the userConfig which requires a bunch of stuff created after the config is initialized. Nevertheless, I was able to work in giving precedence to `skipAnalytics` when passed as a 'constructor' argument